src/hardware/input/mouse_config.cpp

```diff
@@ -146,14 +170,60 @@ static void SetPs2Model(const std::string_view model_str)
 	}
 }
 
-static void SetComModel(const std::string_view model_str)
+static void set_serial_mouse_model(const std::string_view model_str)
 {
 	[[maybe_unused]] const auto result = MouseConfig::ParseComModel(
 	        model_str, mouse_config.model_com, mouse_config.model_com_auto_msm);
 	assert(result);
 }
 
-static void SetSensitivity(const std::string_view sensitivity_str)
+static void set_dos_driver_options(const std::string_view options_str)
+{
+	auto& last_options_str = mouse_config.dos_driver_last_options_str;
+	if (last_options_str == options_str) {
+		return;
+	}
+	last_options_str = options_str;
+
+	mouse_config.dos_driver_immediate = false;
+	mouse_config.dos_driver_modern    = false;
+
+	bool config_needs_sync = false;
+
+	for (const auto& token : split(options_str, " ,")) {
+		if (token == "immediate") {
+			mouse_config.dos_driver_immediate = true;
+		} else if (token == "modern") {
+			mouse_config.dos_driver_modern = true;
+		} else {
+			LOG_WARNING(
+			        "MOUSE: Invalid 'builtin_dos_mouse_driver_options' "
+			        "parameter: '%s', ignoring",
+			        token.c_str());
+			config_needs_sync = true;
+		}
+	}
+
+	if (config_needs_sync) {
```

Comment:
Hmm, bit of an overkill, isn't it? Generally, config settings just revert to the default if an invalid setting provided.

So you could just do this:

```cpp
secprop.Add_string("builtin_dos_mouse_driver_options", always, DefaultBuiltinDosMouseDriverOptions);
```

Then you can ditch the whole `config_needs_sync` thing and simply do this after `LOG_WARNING`:
```cpp
		set_section_property_value("mouse",
		                           "builtin_dos_mouse_driver_options",
		                           DefaultBuiltinDosMouseDriverOptions);
```

Reply:
OK, simplified handling.